pymongo/mongo_client.py

```diff
@@ -1192,6 +1193,9 @@ def _select_server(self, server_selector, session, address=None):
           - `address` (optional): Address when sending a message
             to a specific server, used for getMore.
         """
+        if self._is_closed:
+            raise InvalidOperation("Once a MongoClient is closed, "
+                                   "all operations raise an error")
```

Comment:
It might make more sense to put this logic in `Topology.open()` that way we will never reopen a closed Topology.